aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt

```diff
@@ -440,12 +440,44 @@ open class AztecText : AppCompatEditText, TextWatcher, UnknownHtmlSpan.OnUnknown
     // Setup the keyListener(s) for Backspace and Enter key.
     // Backspace: If listener does return false we remove the style here
     // Enter: Ask the listener if we need to insert or not the char
-    private fun setupBackspaceAndEnterDetection() {
+    private fun setupKeyListenersAndInputFilters() {
         //hardware keyboard
         setOnKeyListener { _, _, event ->
             handleBackspaceAndEnter(event)
         }
 
+        // This InputFilter created only for the purpose of avoiding crash described here:
+        // https://android-review.googlesource.com/c/platform/frameworks/base/+/634929
+        // https://github.com/wordpress-mobile/AztecEditor-Android/issues/729
+        // the rationale behind this workaround is that the specific crash happens only when adding/deleting text right
+        // before an AztecImageSpan, so we detect the specific case and re-create the contents only when that happens.
+        // This is indeed tackling the symptom rather than the actual problem, and should be removed once the real
+        // problem is fixed at the Android OS level as described in the following url
+        // https://android-review.googlesource.com/c/platform/frameworks/base/+/634929
+        val dynamicLayoutCrashPreventer = InputFilter { source, start, end, dest, dstart, dend ->
+            var temp = source
+            if (dest.length > dend+1) {
+                // if there are any images right after the destination position, hack the text
+                val spans = dest.getSpans(dstart, dend+1, AztecImageSpan::class.java)
+                if (spans.isNotEmpty()) {
+                    // test: is this an insertion?
+                    if (dstart == dend) {
```

Comment:
Is it possible to move this check before ` val spans = dest.getSpans(dstart, dend+1, AztecImageSpan::class.java)` ?
It should improve performances a bit, considering that this filter is called at any key pressed.

Reply:
gotcha, nice improvement, made change as suggested in 1ff08ac1